### PR TITLE
feat(glossary): updated glossary

### PR DESCRIPTION
**Pull request by EmmanuelOluwafemi**
## Description

Hey guys, I updated the glossary with these five words: “Proxy,” “HttpProvider,” “WebSocketProvider,” “Events,” and “Logs.” I was talking to @SantiagoDevRel, and we came up with this idea.

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: docs/docs/glossary/index.md [hunk 1]
```diff
@@ -108,3 +108,85 @@ contract Test {
...
+```
+
+## Event logs
+
+`Logs` in Web3.js are a part of **Ethereum transactions** that contain **information about events triggered** within smart contracts. They provide a way to record and retrieve significant occurrences within the blockchain. `Event logs` are particularly useful for tracking changes, and debugging.
+
+#### Example
+
+```javascript title='Event Log in web3.js'
+const Web3 = require('web3');
```
Comment (SantiagoDevRel) [inline docs/docs/glossary/index.md]: let's use ES for all imports
```ts
import { Web3 } from 'web3'
```

File: docs/docs/glossary/index.md [hunk 2]
```diff
@@ -108,3 +108,85 @@ contract Test {
...
+
+  function transfer(address recipient, uint amount) public {
+    // ... transfer logic ...
+    emit Transfer(msg.sender, recipient, amount);
+  }
+}
+```
+
+```javascript title='Event in web3.js'
+const Web3 = require('web3');
```
Comment (SantiagoDevRel) [inline docs/docs/glossary/index.md]: let's use ES for all imports
```ts
import { Web3 } from 'web3'
```

File: docs/docs/glossary/index.md [hunk 3]
```diff
@@ -108,3 +108,85 @@ contract Test {
...
+```javascript title='HttpProvider example'
+const Web3 = require('web3');
+const web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:8545'));
+```
+
+## WebSocketProvider
+`WebSocketProvider` in Web3.js connects your application to an Ethereum node via WebSocket, enabling real-time and asynchronous communication. This provider is ideal for applications needing real-time updates, such as new blocks or smart contract events. It offers better performance for high-throughput applications compared to `HttpProvider`. Ensure secure connections with `wss://` for exposed endpoints. Handle reconnections gracefully for reliable operation.
+
+```javascript title='WebSocketProvider example'
+const Web3 = require('web3');
```
Comment (SantiagoDevRel) [inline docs/docs/glossary/index.md]: let's use ES for all imports
```ts
import { Web3 } from 'web3'
```

File: docs/docs/glossary/index.md [hunk 4]
```diff
@@ -108,3 +108,85 @@ contract Test {
...
+
+## Proxy
+
+A `proxy` in Web3.js serves as an intermediary between your application and an Ethereum node, **facilitating communication** by **forwarding requests and responses**. Configuring a proxy can help overcome network restrictions, enhance security, and improve load balancing. Understanding how to set up and use proxies with `HttpProvider` and `WebsocketProvider` in Web3.js is crucial for building robust and secure blockchain applications.
+
+## HttpProvider
+`HttpProvider` in Web3.js connects an application to an Ethereum node over HTTP. It allows for sending transactions, reading blockchain data, and interacting with smart contracts. You create a Web3 instance with the node’s URL to establish the connection. It’s essential for DApps needing blockchain interaction but can block the event loop, so alternatives like `WebSocketProvider` might be used for better performance. Proper security measures are crucial when exposing HTTP endpoints.
+
+```javascript title='HttpProvider example'
+const Web3 = require('web3');
```
Comment (SantiagoDevRel) [inline docs/docs/glossary/index.md]: let's use ES for all imports
```ts
import { Web3 } from 'web3'
```

**Review by SantiagoDevRel — CHANGES_REQUESTED** [review]

**Review by SantiagoDevRel — APPROVED** [review]

**Comment by luu-alex** [comment]
@danforbes

File: docs/docs/glossary/index.md [hunk 5]
```diff
@@ -108,3 +108,85 @@ contract Test {
...
 ```
+
+## Proxy
+
+A `proxy` in Web3.js serves as an intermediary between your application and an Ethereum node, **facilitating communication** by **forwarding requests and responses**. Configuring a proxy can help overcome network restrictions, enhance security, and improve load balancing. Understanding how to set up and use proxies with `HttpProvider` and `WebsocketProvider` in Web3.js is crucial for building robust and secure blockchain applications.
+
+## HttpProvider
+`HttpProvider` in Web3.js connects an application to an Ethereum node over HTTP. It allows for sending transactions, reading blockchain data, and interacting with smart contracts. You create a Web3 instance with the node’s URL to establish the connection. It’s essential for DApps needing blockchain interaction but can block the event loop, so alternatives like `WebSocketProvider` might be used for better performance. Proper security measures are crucial when exposing HTTP endpoints.
+
+```javascript title='HttpProvider example'
```
Comment (luu-alex) [inline docs/docs/glossary/index.md]:
```suggestion
```typescript
```

File: docs/docs/glossary/index.md [hunk 6]
```diff
@@ -108,3 +108,85 @@ contract Test {
...
+});
+```
+
+## Event logs
+
+`Logs` in Web3.js are a part of **Ethereum transactions** that contain **information about events triggered** within smart contracts. They provide a way to record and retrieve significant occurrences within the blockchain. `Event logs` are particularly useful for tracking changes, and debugging.
+
+#### Example
+
+```javascript title='Event Log in web3.js'
```
Comment (luu-alex) [inline docs/docs/glossary/index.md]:
```suggestion
```typescript
```

File: docs/docs/glossary/index.md [hunk 7]
```diff
@@ -108,3 +108,85 @@ contract Test {
     }
 ]
 ```
+
+## Proxy
```
Comment (jdevcs) [inline docs/docs/glossary/index.md:112]: @danforbes what do you suggest for keeping glossary section?

File: docs/docs/glossary/index.md [hunk 8]
```diff
@@ -108,3 +108,85 @@ contract Test {
...
+
+A `proxy` in Web3.js serves as an intermediary between your application and an Ethereum node, **facilitating communication** by **forwarding requests and responses**. Configuring a proxy can help overcome network restrictions, enhance security, and improve load balancing. Understanding how to set up and use proxies with `HttpProvider` and `WebsocketProvider` in Web3.js is crucial for building robust and secure blockchain applications.
```
Comment (danforbes) [inline docs/docs/glossary/index.md]: I think we need to be careful about saying that setting something up properly is crucial for security without providing any additional context or information about how to actually set things up so they are secure. Can we link to some documentation that explains how to securely configure a proxy?

File: docs/docs/glossary/index.md [hunk 9]
```diff
@@ -108,3 +108,85 @@ contract Test {
...
+
+## HttpProvider
+`HttpProvider` in Web3.js connects an application to an Ethereum node over HTTP. It allows for sending transactions, reading blockchain data, and interacting with smart contracts. You create a Web3 instance with the node’s URL to establish the connection. It’s essential for DApps needing blockchain interaction but can block the event loop, so alternatives like `WebSocketProvider` might be used for better performance. Proper security measures are crucial when exposing HTTP endpoints.
```
Comment (danforbes) [inline docs/docs/glossary/index.md]: The use of the word "essential" is confusing to me here. What are you trying to say? I would also be careful about bringing up the idea of crucial security measures without providing additional context about the specific measures that need to be taken.

**Review by danforbes — COMMENTED** [review]
I've left a few comments. In general, I think it would be good for the glossary definitions to include links to the relevant sections of the documentation.

File: docs/docs/glossary/index.md (hunk 7 above)
Comment (danforbes) [reply docs/docs/glossary/index.md:112]: I think a good glossary can be extremely helpful, but it's important that it's managed carefully.

Comment (EmmanuelOluwafemi) [reply docs/docs/glossary/index.md:112]: Thanks for the feedback, I will work on this

File: docs/docs/glossary/index.md (hunk 9 above)
Comment (EmmanuelOluwafemi) [reply docs/docs/glossary/index.md]: Thanks for the feedback, I will work on this

**Comment by EmmanuelOluwafemi** [comment]
Hi @danforbes, I have updated the pr based on your feedbacks

**Review by danforbes — APPROVED** [review]
Thank you @EmmanuelOluwafemi 🙏🏻

**Pull request merged**